### PR TITLE
Refactor of API class into UrlResolver and Utilities classes

### PR DESCRIPTION
**Pull request by harej**
This pull request:
* Refactors URL resolving methods (i.e., "is this link a Wayback Machine link") into a new class called UrlResolver
* Creates a new Utilities class that currently only has two methods and will have more added soon
* Fixes the current acceptance testing and writes new unit testing for UrlResolver methods

### CODE REVIEW

File: app/src/Core/APII.php [hunk 1]
```diff
@@ -2872,94 +2863,101 @@ public function retrieveArchive( $data )
...
+				"archive.li",
+				"archive.vn",
+				"archive.md",
+				"archive.ph"];
+
+		if ( self::hostMatch( $parts['host'], [ "europarchive.org", "internetmemory.org" ] ) ) {
+			$resolvedData = UrlResolver::resolveEuropa( $url );
+		} elseif ( self::hostMatch( $parts['host'], [ "webarchive.org.uk" ] ) ) {
+			$resolvedData = UrlResolver::resolveUKWebArchive( $url );
+		} elseif ( self::hostMatch( $parts['host'], [ "archive.org", "waybackmachine.org" ] ) ) {
```
Comment (cyberpower678) [inline app/src/Core/APII.php:2896]: It should be web.archive.org.  Anything outside of it is not the Wayback Machine, and may be misinterpreted.  Try to be as specific as possible when it comes to domains.

Comment (cyberpower678) [reply app/src/Core/APII.php:2896]: I note the old code didn't have it either, which is actually a bug. :p

**Review by cyberpower678 — CHANGES_REQUESTED** [review]
The CURL handles will not initialize the way they are coded right now.  This will result in a fatal runtime error most likely.

I'm wondering why many test cases are commented out, and one of the acceptance result looks off.

**Comment by decodedfile** [comment]
Duplicate of #

**Pull request closed without merging**